rpcd/playbooks/roles/elasticsearch/templates/upgrade-pre-flight-checks.py.j2

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import apt
+import sys
+
+cache = apt.Cache()
+es_pkg = cache['elasticsearch']
+
+es_version = es_pkg.installed.version
+
+if es_pkg.installed.version == '{{ elasticsearch_version }}':
```

Comment:
I'm wondering if using exit codes is the best way to manage this. Can't ansible understand it if you print JSON, something like:

```py
if es_pkg.installed.version == '{{ elasticsearch_version }}':
    pkg_status = {
        'status': 'latest-installed',
        'explanation': 'Latest version of elasticsearch installed already. Skip upgrade.',
   }
elif es_pkg.installed_version == '{{ elasticsearch_reindex_version }}':
    pkg_status = {
        'status': 'reindexing',
        'explanation': '...',
    }
else:
    pkg_status = {
        'status': 'needs-upgrade',
        'explanation': 'Elasticsearch is at version {} and needs to be upgraded'.format(es_pkg.installed.version),
    }

print(json.dumps(pkg_status))
```

That said, I'm not sure how ansible detects/parses that, so you may be better off with this method for now and then experimenting on improving this for 13.2